boa_gc/src/internals/gc_box.rs

```diff
@@ -0,0 +1,195 @@
+use crate::Trace;
+use std::cell::Cell;
+use std::fmt;
+use std::ptr::{self, NonNull};
+
+// Age and Weak Flags
+const MARK_MASK: usize = 1 << (usize::BITS - 2);
+const WEAK_MASK: usize = 1 << (usize::BITS - 1);
+const ROOTS_MASK: usize = !(MARK_MASK | WEAK_MASK);
+const ROOTS_MAX: usize = ROOTS_MASK;
```

Comment:
Why are these `usize` instead of `u8`? Also, have you considered using `bitflags!{}`?

Reply:
These are `usize` because roots is the counter for rooted objects. I think HalidOdat mentioned it above, but at `usize::BITS` that would allow us in worse case to have 2^30 roots available before a panic. If we make this a `u8`, then each header will only be able to have at best 2^8 roots.

I'm not too familiar with `bitflags!{}`. I'll take a look at it. Is there a benefit to using it that you had in mind?

Reply:
Nope, I thought this was just a flag, not a counter. bitflags is nice if you want to have binary flags in a u8. This is good :)